### PR TITLE
Updated Engine's schemf module to support array paths

### PR DESCRIPTION
**Pull request by JavierBejMen**
|Related issue|
|---|
|#22679|

This PR adds to the schemf module the ability to query array items types of schema fields that are arrays.

### CODE REVIEW

File: src/engine/source/schemf/src/schema.cpp [hunk 1]
```diff
@@ -118,16 +137,48 @@ const Field& Schema::get(const DotPath& name) const
...
-            return false;
+            if (!isParentSchema)
+            {
+                return false;
+            }
+
+            throw std::runtime_error(fmt::format("Field '{}' does not exist in '{}'", it->data(), name.str()));
         }
 
+        isParentSchema = true;
```
Comment (NahuFigueroa97) [inline src/engine/source/schemf/src/schema.cpp:154]: question: Why do you set this variable and then not use it?

File: src/engine/source/schemf/src/schema.cpp [hunk 2]
```diff
@@ -98,7 +98,26 @@ const Field& Schema::get(const DotPath& name) const
 
         if (it != name.cend() - 1)
         {
-
+            // Handle arrays e.g. "field/0"
+            // If the field is an array and the next part is the last one and a number, return a new field with the
+            // array type
+            if (entry->second.isArray() && (it + 1) == (name.cend() - 1))
```
Comment (NahuFigueroa97) [inline src/engine/source/schemf/src/schema.cpp:104]: improve-desing: I notice that the first part of this block is repeated for both get() and has_field(), could you modulate it a little more?

File: src/engine/source/schemf/test/src/component/schema_test.cpp [hunk 3]
```diff
@@ -141,3 +149,18 @@ INSTANTIATE_TEST_SUITE_P(SchemaTest,
                                            LoadTuple(R"({"a": {}})", false),
                                            LoadTuple(R"({"a": [{"type": "text"}]})", false),
                                            LoadTuple(R"([{"a": {"type": "keyword"}}])", false)));
+
+TEST(SchemaTest, ArrayItem)
```
Comment (NahuFigueroa97) [inline src/engine/source/schemf/test/src/component/schema_test.cpp:153]: Is this a particular case? If it is just a case, could you document the test?

File: src/engine/source/schemf/src/schema.cpp (hunk 1 above)
Comment (JavierBejMen) [reply src/engine/source/schemf/src/schema.cpp:154]: It is used

File: src/engine/source/schemf/test/src/component/schema_test.cpp (hunk 3 above)
Comment (JavierBejMen) [reply src/engine/source/schemf/test/src/component/schema_test.cpp:153]: I think the test is self documented. It just tests array items get and exists.

File: src/engine/source/schemf/src/schema.cpp (hunk 2 above)
Comment (JavierBejMen) [reply src/engine/source/schemf/src/schema.cpp:104]: The little details makes it not so simple to be the same for both, and it its only used on those two functions.

**Review by NahuFigueroa97 — APPROVED** [review]
LGTM!

**Review by juliancnn — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**